Poke/App.tsx

```diff
@@ -1,24 +1,80 @@
-import React from 'react';
-import {SafeAreaView, StyleSheet, ScrollView, View, Text, StatusBar} from 'react-native';
+import React, {useState} from 'react';
+import {SafeAreaView, StyleSheet, ScrollView, View, Text, StatusBar, Alert} from 'react-native';
 
 import {Colors} from 'react-native/Libraries/NewAppScreen';
 
-import {EmailInput, SenhaInput} from './src/components/login-input';
+import {EmailInput, PasswordInput} from './src/components/login-input';
 import {SubmitButton} from './src/components/submit-button';
+import {validation} from './src/validation';
+import {ApolloClient, InMemoryCache, gql} from '@apollo/client';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
+const client = new ApolloClient({
+  uri: 'https://tq-template-server-sample.herokuapp.com/graphql',
+  cache: new InMemoryCache()
+});
+
+// Email: admin@taqtile.com.br
+// Password: 1234qwer
+
+const login = (email: string, password: string) => {
+   client.mutate({
+    mutation: gql`
+      mutation {
+        login (data:{
+          email: "${email}"
+          password: "${password}"
+        }){
+          token
+        }
+      }
+    `
+  })
+  .then((result) => {
+    console.log(result)
+    const jsonString = JSON.stringify(result);
+    const data = JSON.parse(jsonString);
+    storeData(data.data.login.token);
+  })
+  .catch(err => {
+    const errorString = JSON.stringify(err)
+    const error = JSON.parse(errorString)
+    console.log(error)
+    Alert.alert(error.message)
+  })
+}
+
+const storeData = async (value: string) => {
+  try {
+    await AsyncStorage.setItem('@storage_Key', value)
+    Alert.alert("Nice")
+  } catch (e) {
+    Alert.alert(e)
+  }
+}
 
 const App = () => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const handleSubmit = () => {
+    const validationError = validation(email, password);
+    if(validationError === 'Tudo certo'){
+      login(email, password)
+    }
+    else{
+      Alert.alert(validationError);
+    }
+  };
   return (
     <>
       <StatusBar barStyle="dark-content" />
       <SafeAreaView>
         <ScrollView contentInsetAdjustmentBehavior="automatic" style={styles.scrollView}>
           <View style={styles.body}>
             <Text style={styles.simple}>Bem vindo(a) à Taqtile!</Text>
-            
-            <EmailInput />
-            <SenhaInput />
-            <SubmitButton />
-
+            <EmailInput text={email} onTextChange={setEmail} />
+            <PasswordInput text={password} onTextChange={setPassword} />
+            <SubmitButton onTap={() => {handleSubmit()}} />
```

Comment:
Aqui dá pra passar só a referência da função em vez de criar outra:
```suggestion
            <SubmitButton onTap={handleSubmit} />
```